### DISCUSSION TITLE
Possible extractor regression

### DISCUSSION THREAD

**Discussion by kellytk** [tokio-rs/axum#583] [Q&A] [answered]
Using [ServeFile](https://docs.rs/tower-http/latest/tower_http/services/fs/struct.ServeFile.html) from `tower-http`, with a simple app:

````
let app = Router::new()
    .route("/file.txt", get(file_handler))
    .layer(AddExtensionLayer::new(Arc::new(State {})));
````

and handler signature:

````
async fn file_handler (
    request: Request<Body>,
    shared: extract::Extension<Arc<State>>,
) -> Result<Response<ServeFileSystemResponseBody>, (StatusCode, String)>
````

requests return an "Extensions taken by other extractor" 500 internal server error.  However, if I transpose the `request` and `shared` parameters, the request is fulfilled as expected.  IIRC it worked with either order until recently.

**Comment by davidpdrsn** [comment, marked as answer]
I just tested and it didn't work on 0.3.x and its never supposed to work so everything is normal.

It isn't supported because `Request<Body>` used an extractors cannot be combined with other unless `Request<Body>` is the very last extractor. More details [here](https://docs.rs/axum/0.3.4/axum/extract/index.html#applying-multiple-extractors). Note this links to version 0.3.4 but this behavior has not changed in 0.4.
